### DISCUSSION TITLE
Framework improvement

### DISCUSSION THREAD

**Discussion by ghost** [andialbrecht/sqlparse#736] [Ideas]
Greetings to all!) 

I would like to help the project, and therefore I want to propose my action plan for approval by the main contributors. I want to add typing to this framework to improve the quality of the code and reduce errors, I also want to add black and isort dependencies in order to sort imports and code by one template, and so that the code becomes even a little clearer. I also saw that there are too many errors when running the flake8 check (tox -e flake8), I would like to fix them too. If the main contributors give me permission for the improvements/updates that I have described here, then I would be happy to add them to the code in a couple of months) 

I will be glad to receive feedback and cooperation!)

**Comment by andialbrecht** [comment]
Hi @ashm-tech,

thanks for your interest in improving sqlparse! You are mentioning quite a lot of options, it would be good to focus on one to get started.

Regarding black (and isort) I'm somewhat reserved. IMO it's not that hard to keep code in shape and the last time I've had a look at black it made some formatting decisions I didn't like.
Similar for isort, we don't have that much imports and most of the import blocks should be already sorted. Maybe in a bigger project it would make sense. But from my experience, in bigger projects there might be reasons to use some kind of, let's say, project specific sorting. I'm not sure if that is supported by isort.
I'm happy to hear other opinions, but there should be a real benefit for the project. 

I was a bit surprised that you mentioned a lot of flake8 errors, because flake8 is checked in the CI pipeline: https://github.com/andialbrecht/sqlparse/actions/runs/6222339438/job/16886085504 
But you're right, in the CI pipeline only the module itself is checked. When running `tox -e flake` the tests are also checked and they're producing some style errors. That would be something worth addressing.

Andi
